lib/helper.js

```diff
@@ -6,16 +6,28 @@ const Hoek = require('@hapi/hoek');
  * Convert job steps from array to object for faster lookup
  * @method convertFromArrayToObject
  * @param  {Array} steps Job step array of objects
- * @return {Object}      Job object
+ * @return {Object}      Object with: job object, array of lockedStepNames
  */
 function convertFromArrayToObject(steps) {
-    return steps.reduce((obj, item) => {
+    const lockedStepNames = [];
+    const stepObj = steps.reduce((obj, item) => {
         const key = Object.keys(item)[0];
-
-        obj[key] = item[key];
+        const substepKeys = typeof item[key] === 'object' ? Object.keys(item[key]) : undefined;
+
+        // Compress step if only has command key
+        if (substepKeys && substepKeys.length === 1) {
+            obj[key] = item[key].command;
```

Comment:
things we can do in javascript , if we we ever switch to typescript it will be some interesting work.